glommio/src/lib.rs

```diff
@@ -364,6 +364,7 @@ pub mod io;
 mod multitask;
 mod networking;
 mod pollable;
+pub mod read_write_lock;
```

Comment:
That means users will import `glommio::read_write_lock::ReadWriteLock` right?

That's suboptimal, I think. It is best if modules are a collected of related things.
There are two options:
1. Don't use a module, and just export `ReadWriteLock` as we do with the `Semaphore`.
2. Put this, together with the `Semaphore` in their own module. People in the past suggested `sync` for this. Now that we have a `Semaphore` and a rwlock it makes sense. This breaks API but I won't be too concerned with that until we reach 1.0. Best to get things right!

Reply:
I suggest adding `pub mod sync;` for all synchronization primitives, following std. 

Though, `std::sync` is a bit of a kitchen sink (pun not intended) and there are plans to, eg, uplift `atomic` to top level, move `Arc` to `rc`, etc.

Reply:
I will move to the sync module then.

Reply:
Thanks! If you are familiar with git black magic, it'd be slightly better to first have a patch that moves the semaphore into sync, and then add this on top.